src/mmw/js/src/data_catalog/models.js

```diff
@@ -380,6 +382,41 @@ var Result = Backbone.Model.extend({
         return response;
     },
 
+    fetchHydroshareDetails: function() {
+        if (this.fetchPromise && !this.get('error')) {
+            return this.fetchPromise;
+        }
+
+        var self = this,
+            id = self.get('id'),
+            setSuccess = function() { self.set('error', false); },
+            setError = function() { self.set('error', true); },
+            startFetch = function() { self.set('fetching', true); },
```

Comment:
May also want to set `error` false.

Reply:
I like this interaction:

![2017-10-24 20 45 39](https://user-images.githubusercontent.com/1430060/31974966-6893bd48-b8fc-11e7-8342-a31416096560.gif)

where you see both the error icon from last time and the loading spinner together. If the new fetch is successful, error will be set to false [here](https://github.com/WikiWatershed/model-my-watershed/pull/2424/files#diff-9521e1449442569cfea57cd15af8f568R413).

Reply:
I think the error icon within the loading spinner looks a little odd/unconventional. Tagging @jfrankl for visual review though.

Reply:
I agree with @ajrobbins, I don't think we should display the spinner and the error icon at the same time.

Reply:
The rest of this looks good to me.

Reply:
@rajadain if you hover over one of the red error icons, do you get a tooltip that explains what that icon means? If not, we should add one.

Reply:
Yes, we do. 

![image](https://user-images.githubusercontent.com/1430060/32074810-8bb90888-ba68-11e7-85f8-7ac7a371b975.png)

I'll remove the error icon when beginning a search.

Reply:
That looks great!